sdk/src/evaluations/index.ts

```diff
@@ -0,0 +1,82 @@
+import type { TypedDataSigner } from "@ethersproject/abstract-signer";
+import { ethers } from "ethers";
+import { CIDString } from "nft.storage";
+
+import { DEFAULT_CHAIN_ID } from "../constants.js";
+import HypercertsStorage from "../storage.js";
+import { StorageError } from "../types/errors.js";
+import { EASEvaluation, EvaluationSource, HypercertEvaluationSchema, IPFSEvaluation } from "../types/evaluation.js";
+import EasEvaluator from "./eas.js";
+
+const EASContractAddress = "0xC2679fBD37d54388Ce493F1DB75320D236e1815e"; // Sepolia v0.26
+
+type HypercertEvaluatorConfig = {
+  chainId: number;
+  address: string;
+  signer: ethers.Signer & TypedDataSigner;
+  storage: HypercertsStorage;
+};
+
+export interface EvaluatorInterface {
+  submitEvaluation: (evaluation: HypercertEvaluationSchema) => Promise<CIDString>;
+}
+
+export default class HypercertEvaluator implements EvaluatorInterface {
+  signer: ethers.Signer & TypedDataSigner;
+  storage: HypercertsStorage;
+  eas: EasEvaluator;
+
+  constructor({
+    config = {
+      chainId: DEFAULT_CHAIN_ID,
+      address: EASContractAddress,
+      signer: new ethers.VoidSigner(""),
+      storage: new HypercertsStorage({}),
+    },
+  }: {
+    config?: HypercertEvaluatorConfig;
+  }) {
+    this.signer = config.signer;
+    this.storage = config.storage;
+    this.eas = new EasEvaluator({
+      config: {
+        address: config.address,
+        chainId: config.chainId,
+        signer: this.signer,
+      },
+    });
+  }
+
+  submitEvaluation = async (evaluation: HypercertEvaluationSchema): Promise<CIDString> => {
+    if (this.storage.readonly) {
+      throw new StorageError("Storage is in readonly mode");
+    }
+
+    let evaluationToStore: HypercertEvaluationSchema | undefined;
+
+    if (isEasEvaluation(evaluation.evaluationSource)) {
```

Comment:
I guss I'm a little confused by the difference between EasEvaluation and IPFS evaluation?

Because EAS supports both on-chain and off-chain evaluations. So EasEvaluation is also off-chain, storing the data on IPFS. I'm okay with not worrying about on-chan evaluations for now. But does that mean the only difference is that IPFSEvaluations are unsigned? Is there any reason not to sign?

Reply:
Good point. There's almost no difference except for the signing. Maybe I scope creeped in IPFS, using EAS with signed data feels a lot more cleaner and valuable to me for this specific usecase. Shall I take out IPFS?

Reply:
pushed

Reply:
see [evaluator/index.ts](https://github.com/hypercerts-org/hypercerts/blob/1a45f6dc37a7e31f49305f610f035eec24e7452e/sdk/src/evaluations/index.ts#LL45C17-L45C17)